src/AspNet.Security.OpenIdConnect.Extensions/OpenIdConnectConstants.cs

```diff
@@ -22,22 +22,61 @@ public static class Algorithms {
         }
 
         public static class Claims {
-            public const string Active = "active";
-            public const string Address = "address";
-            public const string AtHash = "at_hash";
-            public const string ClientId = "client_id";
-            public const string Confidential = "confidential";
-            public const string EmailVerified = "email_verified";
-            public const string PhoneNumber = "phone_number";
-            public const string PhoneNumberVerified = "phone_number_verified";
-            public const string PreferredUsername = "preferred_username";
-            public const string Profile = "profile";
-            public const string Scope = "scope";
-            public const string TokenId = "token_id";
-            public const string TokenType = "token_type";
-            public const string Usage = "usage";
-            public const string Username = "username";
-            public const string Website = "website";
+            public static class Protocol {
+                public const string AtHash = "at_hash";
+                public const string Audience = "aud";
+                public const string AuthorizedParty = "azp";
+                public const string Confidential = "confidential";
+                public const string CryptographicHash = "c_hash";
```

Comment:
> That's fine, but what happens if someone wants to target identity or access token destinations based on the claim types in, say, a UserManager? While not required, having the constants is definitely helpful.

Not sure to understand the concrete use case you describe, but one thing is sure: I don't want to encourage people to use this const class for user claims: people should just use the de facto WS-Fed claims exposed by `ClaimTypes` and rely on the automatic claims mapping. This allows decoupling the app code from the specific token type instead of introducing JWT specific claims in the app code.

> It can be changed in the next version, sure, but since it was already done, I'm not sure why not to use it. :confused:
I think maybe you just like to modify PR code. :laughing: 

Haha :smile: 

Actually no, I just tend to apply the same rule as the ASP.NET team: the status quo always wins, unless we're totally sure the new approach is better than the old one.

Here, I'm not sure the suggested groups were better/clearer than the flat approach. E.g what's the point of separating the claims exposed by the protocol class from the ones defined in the introspection class? The JWT claims set returned by the introspection endpoint contains both protocol claims and user attributes so why introducing a distinction?